distributor/cmd/internal/distributor/distributor_test.go

```diff
@@ -59,6 +66,98 @@ var (
 	}
 )
 
+var helper dbHelper
+
+type dbHelper struct {
+	address string
+	nextDB  uint32
+}
+
+func (h *dbHelper) connect() (*sql.DB, error) {
+	connectString := fmt.Sprintf("root:secret@(%s)/mysql", h.address)
+	return sql.Open("mysql", connectString)
+}
+
+func (h *dbHelper) create(testName string) (*sql.DB, error) {
+	db, err := h.connect()
```

Comment:
Should there be a `defer db.Close()` since we re-open the DB with a more specific URI below?

Reply:
Done.